instrumentation/active_support/lib/opentelemetry/instrumentation/active_support/span_subscriber.rb

```diff
@@ -30,19 +30,21 @@ def self.subscribe(
 
         subscriber_object = ::ActiveSupport::Notifications.subscribe(pattern, subscriber)
 
-        ::ActiveSupport::Notifications.notifier.synchronize do
-          subscribers = ::ActiveSupport::Notifications.notifier.instance_variable_get(:@string_subscribers)[pattern]
-
-          if subscribers.nil?
-            OpenTelemetry.handle_error(
-              message: 'Unable to move OTEL ActiveSupport Notifications subscriber to the front of the notifications list which may cause incomplete traces.' \
-                       'Please report an issue here: ' \
-                       'https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/new?labels=bug&template=bug_report.md&title=ActiveSupport%20Notifications%20subscribers%20list%20is%20nil'
-            )
-          else
-            subscribers.unshift(
-              subscribers.delete(subscriber_object)
-            )
+        if Gem::Version.new(::Rails.version) < Gem::Version.new('7.2')
```

Comment:
Is this disabling forcing the subscriber to the from of the list just for Rails 7.2? 

Is there something available in 7.2 that allows us to force this subscriber to be inserted in the front or that not possible because the notification order non-deterministic?

Reply:
From my understanding, you shouldn't need to force the subscriber to be first anymore after Rails 7.2, because  the new changes in Rails guarantee that if an exception occurs within a subscriber the remaining subscribers will still be run. I believe this whole code block was written to work around that problem

Reply:
@arielvalentin by design there isn't anything available in any version of Rails which allows you to do this, and it will not be added. Everyone thinks their subscriber is the most important and wants it to go first 😅.